compiler/qsc_eval/src/lib.rs

```diff
@@ -400,6 +400,16 @@ impl Env {
             .collect();
         variables_by_scope.into_iter().flatten().collect::<Vec<_>>()
     }
+
+    #[must_use]
+    pub fn is_empty(&self) -> bool {
```

Comment:
I think `is_empty` can be removed. It's not called anywhere, and because of how `leave_scope` never pops the last scope created by the `default` implementation, `is_empty` actually always returns false, so it will likely never be useful.

Reply:
I added it because of a clippy error but it makes more sense to not have it so I removed it.